tests/backup/backup_restore_basic_test.go

```diff
@@ -2800,3 +2801,208 @@ var _ = Describe("{BackupRestoreOnDifferentK8sVersions}", func() {
+			log.InfoD("Validating app namespaces in source cluster")
+			ValidateApplications(contexts)
+		})
+		Step("Create schedule backup", func() {
+			log.InfoD("Creating a schedule backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			scheduleName = fmt.Sprintf("%s-schedule-%v", BackupNamePrefix, time.Now().Unix())
+			labelSelectors := make(map[string]string)
+			err = CreateScheduleBackup(scheduleName, SourceClusterName, backupLocationName, backupLocationUID, appNamespaces,
+				labelSelectors, orgID, "", "", "", "", schedulePolicyName, schedulePolicyUid, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of schedule backup with schedule name [%s]", scheduleName))
+			firstScheduleBackupName, err := GetFirstScheduleBackupName(ctx, scheduleName, orgID)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching the name of the first schedule backup [%s]", firstScheduleBackupName))
+			scheduleNames = append(scheduleNames, scheduleName)
+		})
+		Step("Delete the App namespaces created", func() {
+			log.InfoD("Delete the App namespaces created")
+			opts := make(map[string]bool)
+			opts[SkipClusterScopedObjects] = true
+			log.InfoD("Destroying apps")
+			for _, ctx := range contexts {
+				TearDownContext(ctx, opts)
+			}
+			for _, namespace := range appNamespaces {
+				err := DeleteAppNamespace(namespace)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verfiying the deletion of namespace - [%s]", namespace))
+			}
+			// Sleeping 10 sec for NameSpace to be terminated.
+			time.Sleep(10 * time.Second)
+		})
+		Step("Recreating the namespaces deleted", func() {
+			log.InfoD("Recreating the namespaces deleted")
+			for i := 0; i < numDeployments; i++ {
+				taskName := fmt.Sprintf("src-%s-%d", taskNamePrefix, i)
+				nameSpace := fmt.Sprintf("test-namespace-%s", taskName)
+				appContexts := ScheduleApplicationsOnNamespace(nameSpace, taskName)
+				contexts = append(contexts, appContexts...)
+				for index, ctx := range appContexts {
+					appName := Inst().AppList[index]
+					ctx.ReadinessTimeout = appReadinessTimeout
+					log.InfoD("Scheduled application [%s] in source cluster in namespace [%s]", appName, nameSpace)
+				}
+			}
+		})
+		Step("Validate app namespaces in source cluster", func() {
+			log.InfoD("Validating app namespaces in source cluster")
+			ValidateApplications(contexts)
+		})
+		Step("Verify deleted application namespaces inclusion in next schedule backup", func() {
+			log.InfoD("Verifying deleted application namespaces inclusion in next schedule backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD("Waiting for %v minutes for the next schedule backup to be triggered", schedulePolicyintervalInMins)
+			time.Sleep(time.Duration(schedulePolicyintervalInMins) * time.Minute)
+			latestScheduleBackupName, err = GetLatestScheduleBackupName(ctx, scheduleName, orgID)
```

Comment:
Actually there is no way for you to know if this the latest scheduled backup. The best way to approach this is like this

```
                        allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching names of all schedule backups of schedule named [%s]", scheduleName))
			currentScheduleBackupCount := len(allScheduleBackupNames)
			log.InfoD("Current number of schedule backups is [%v]", currentScheduleBackupCount)
			nextScheduleBackupOrdinal := currentScheduleBackupCount + 1
			log.InfoD("Ordinal of the next schedule backup is [%v]", nextScheduleBackupOrdinal)
			checkOrdinalScheduleBackupCreation := func() (interface{}, bool, error) {
				ordinalScheduleBackupName, err := GetOrdinalScheduleBackupName(ctx, scheduleName, nextScheduleBackupOrdinal, orgID)
				if err != nil {
					return "", true, err
				}
				return ordinalScheduleBackupName, false, nil
			}
			log.InfoD("Waiting for %v minutes for the next schedule backup to be triggered", intervalInMins)
			time.Sleep(time.Duration(intervalInMins) * time.Minute)
			nextScheduleBackupName, err = task.DoRetryWithTimeout(checkOrdinalScheduleBackupCreation, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second)
			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching next schedule backup name of ordinal [%v] of schedule named [%s]", nextScheduleBackupOrdinal, scheduleName))
			log.InfoD("Next schedule backup name [%s]", nextScheduleBackupName.(string))
			err = backupSuccessCheck(nextScheduleBackupName.(string), orgID, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second, ctx)
			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying success of next schedule backup named [%s] of schedule named [%s]", nextScheduleBackupName.(string), scheduleName))
			nextScheduleBackupUid, err := Inst().Backup.GetBackupUID(ctx, nextScheduleBackupName.(string), orgID)
			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching uid of of next schedule backup named [%s] of schedule named [%s]", nextScheduleBackupName, scheduleName))
			backupInspectRequest := &api.BackupInspectRequest{
				Name:  nextScheduleBackupName.(string),
				Uid:   nextScheduleBackupUid,
				OrgId: orgID,
			}
			resp, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
			dash.VerifyFatal(err, nil, fmt.Sprintf("Inspecting next schedule backup named [%s] and uid [%s]", nextScheduleBackupName, nextScheduleBackupUid))
			backedUpNamespaces := resp.GetBackup().GetNamespaces()
			log.InfoD("Namespaces in next schedule backup named [%s] and uid [%s] are [%v]", nextScheduleBackupName, nextScheduleBackupUid, backedUpNamespaces)
			for _, namespace := range newAppNamespaces {
				dash.VerifyFatal(strings.Contains(strings.Join(backedUpNamespaces, ","), namespace), true, fmt.Sprintf("Checking the new application namespace [%s] against the next scheduled backup named [%s]", namespace, nextScheduleBackupName))
			}
```

Reply:
In fact it is better to make this a function called `func WaitForNextScheduledBackup(ctx context.Context, scheduleName, schedulePolicyintervalInMins)`

Reply:
Please do have a look at GetNextScheduleBackupName in backup_helper.go

Reply:
Made changes to use GetNextScheduleBackupName. Thanks !